xwiki-commons-core/xwiki-commons-diff/xwiki-commons-diff-api/src/main/java/org/xwiki/diff/internal/DefaultDiffManager.java

```diff
@@ -494,19 +518,25 @@
 
         chunkSize = Math.max(deltaCurrent.getMaxChunkSize(), deltaNext.getMaxChunkSize());
 
+        int prevMinIndex = Math.min(deltaCurrent.getPrevious().getIndex(), deltaNext.getPrevious().getIndex());
+        int nextMinIndex = Math.min(deltaCurrent.getNext().getIndex(), deltaNext.getNext().getIndex());
+        int newIndex = index;
         if (deltaCurrent.getType() == Type.INSERT && deltaNext.getType() == Type.INSERT) {
             subsetPrevious = Collections.emptyList();
         } else {
-            int newIndex = Math.min(deltaCurrent.getPrevious().getIndex(), deltaNext.getPrevious().getIndex());
+            newIndex = prevMinIndex;
             int newOffsetEnd = Math.min(chunkSize, previous.size()) + newIndex;
             if (newOffsetEnd > previous.size()) {
                 newOffsetEnd = previous.size();
             }
-            subsetPrevious = new ArrayList<>(previous.subList(newIndex, newOffsetEnd));
+            subsetPrevious = new ArrayList<>(extractFromList(previous, newIndex, newOffsetEnd));
         }
 
-        List<E> subsetNext = new ArrayList<>(extractConflictPart(deltaNext, previous, next, chunkSize, index));
-        List<E> subsetCurrent = new ArrayList<>(extractConflictPart(deltaCurrent, previous, current, chunkSize, index));
+        // We need to always use the minimum index to extract the conflict.
+        newIndex = Math.min(prevMinIndex, nextMinIndex);
+        List<E> subsetNext = new ArrayList<>(extractConflictPart(deltaNext, previous, next, chunkSize, newIndex));
```

Comment:
Could the wrapping in a new list be moved inside extractConflictPart? Seems like it is needed every time this method is called, and it looks error prone.

Reply:
Indeed I improved things.